src/main/java/server/ServerSocketWrapper.java

```diff
@@ -14,17 +22,58 @@ public ServerSocketWrapper(Socket clientSocket, SocketIo socketIo) throws IOExce
     this.socketIo = socketIo;
   }
 
-  public String getRequest() throws IOException {
-    String request = socketIo.read();
-    return (request.isBlank()) ? null : request;
+  public Request getRequest() throws IOException {
```

Comment:
Why you decided to move request parsing to `ServerSocketWrapper`?

Reply:
The solution for the bug prevents having to read bytes `available` by parsing the request and reading the amount of bytes indicated in the content length header to get the body. But now that you mention it I think I'll extract `getRequest()` into its own `RequestReader` class instead of leaving it as a responsibility of `ServerSocketWrapper`

Reply:
I like this idea.

Reply:
updated!